Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -621,7 +621,22 @@ bool LocalFrameViewLayoutContext::pushLayoutState(RenderBox& renderer, const Lay
     
 void LocalFrameViewLayoutContext::popLayoutState()
 {
+    if (!layoutState())
+        return;
+
+    auto currentLineClamp = layoutState()->lineClamp();
+
     m_layoutStateStack.removeLast();
+
+    if (currentLineClamp) {
+        // Propagates the current line clamp state to the parent.
+        if (auto* layoutState = this->layoutState(); layoutState->lineClamp()) {
+            layoutState->setLineClamp(std::make_optional(RenderLayoutState::LineClamp {
+                layoutState->lineClamp()->maximumLineCount,
+                currentLineClamp->currentLineCount,
+            }));
+        }
+    }
```

Comment:
I think this could be collapsed to

```
    if (!layoutState())
        return;

    auto currentLineClamp = layoutState()->lineClamp();

    m_layoutStateStack.removeLast();

    if (auto* layoutState = this->layoutState(); layoutState && layoutState->lineClamp())
        layoutState->setLineClamp(currentLineClamp);

```
maybe asserting on not mutating the maximum line clamp value

Reply:
I chose to assert for `maximumLineCount` since it should be an immutable state established at `applyModernLineClamp`.